internal/database/cache.go

```diff
@@ -22,6 +24,34 @@ type Cache struct {
 	subscription map[string]*SubscriptionDocument
 }
 
+type operationCacheIterator struct {
+	operation map[string]*OperationDocument
+	err       error
+}
+
+func (iter operationCacheIterator) Items(ctx context.Context) iter.Seq[[]byte] {
+	return func(yield func([]byte) bool) {
+		for _, doc := range iter.operation {
+			// Marshalling the document struct only to immediately unmarshal
+			// it back to a document struct is a little silly but this is to
+			// conform to the DBClientIterator interface.
+			item, err := json.Marshal(doc)
+			if err != nil {
+				iter.err = err
+				return
+			}
+
+			if !yield(item) {
+				return
+			}
+		}
+	}
+}
+
+func (iter operationCacheIterator) GetError() error {
+	return iter.err
+}
```

Comment:
this doesn't look concurrency safe - not sure if that's a requirement. Should we at least log `err` as we set it? How would we know what call to Items failed?

Reply:
It's not meant to be concurrency safe, it's meant to be used as part of a [range-over-function](https://go.dev/blog/range-functions) pattern, which is new in Go 1.23.  (Think generator functions in Python, if you're familiar with that.)  The pattern as described in the blog post doesn't address the possibility of iteration failing (such as when paging over incoming data from a remote source) so I came up with this iterator interface as a way to stash an iteration error.

The idiom looks like:
```
for item := range iterator.Items(ctx) {
    ...
}

// Find out if iteration completed or aborted early.
err := iterator.GetError()
```

An iteration error would immediately break the "for" loop.

Reply:
Also maybe worth mentioning, since I didn't realize you were highlighting the in-memory cache...

I don't believe this particular function is currently used.  I wrote it for the in-memory cache in order to fulfill the `DBClient` interface, but the in-memory cache is nowadays only used to mock database operations in unit tests.